src/api/firebase.js

```diff
@@ -144,13 +144,15 @@ export async function createList(userId, userEmail, listName) {
 export async function shareList(listPath, currentUserId, recipientEmail) {
 	// Check if current user is owner.
 	if (!listPath.includes(currentUserId)) {
+		throw new Error('The list is not yours to share.');
```

Comment:
genuine question: Are there best practices about where to validate form input? For example, this check I think could be done either on the front or back end, so are there conventions about which is better?

Reply:
Honestly, I'm not totally sure! I think we went with ease in this case for the form validation. But a savvy user could definitely bypass the client-side as coded if they were so inclined.